[WIP] Added Azure Nova support

sanketsudake opened the pull request:
Change-Id: I0f6ecc0f8a666b35cdd5204571c640a7b309d11b

Commit:
[WIP] Added Azure Nova support

Change-Id: I0f6ecc0f8a666b35cdd5204571c640a7b309d11b

shahpratikr commented on cinder/tests/unit/volume/drivers/azure/data/disk/get_disk.json:
```diff
@@ -0,0 +1,15 @@
+    "id": "/subscriptions/e6cefc59-5a84-4aa3-813f-142adfc86d49/resourceGroups/omni/providers/Microsoft.Compute/disks/volume1",
+    "encryption_settings": null,
+    "owner_id": null
+}
```
this file should not be a JSON file. Response we get when we call REST API, is an object of a class Disk

shahpratikr commented on cinder/tests/unit/volume/drivers/azure/data/snapshot/get_snapshot.json:
```diff
@@ -0,0 +1,15 @@
+    "id": "/subscriptions/e6cefc59-5a84-4aa3-813f-142adfc86d49/resourceGroups/omni/providers/Microsoft.Compute/snapshots/snapshot3",
+    "encryption_settings": null,
+    "owner_id": null
+}
```
same as above

shahpratikr commented on cinder/volume/drivers/azure/config.py:
```diff
@@ -0,0 +1,46 @@
+resource_group = cfg.CONF.AZURE.resource_group
+azure_pool_name = cfg.CONF.AZURE.azure_pool_name
+azure_free_capacity_gb = cfg.CONF.AZURE.azure_free_capacity_gb
+azure_total_capacity_gb = cfg.CONF.AZURE.azure_total_capacity_gb
```
this file is already up for review in my review request. why add again?

shahpratikr commented on nova/virt/azure/config.py:
```diff
@@ -0,0 +1,43 @@
+    cfg.StrOpt('client_id', help='Azure client id'),
+    cfg.StrOpt('client_secret', help='Azure Client secret', secret=True),
+    cfg.StrOpt('subscription_id', help='Azure subscription id'),
+    cfg.StrOpt('zone', help='Azure zone'),
```
it should be region

shahpratikr commented on nova/virt/azure/create-nova-flavors.py:
```diff
@@ -0,0 +1,111 @@
+        client_id = get_env_param('AZURE_CLIENT_ID')
+        client_secret = get_env_param('AZURE_CLIENT_SECRET')
+        subscription_id = get_env_param('AZURE_SUBSCRIPTION_ID')
+        self.zone = get_env_param('AZURE_ZONE')
```
should be region

shahpratikr commented on nova/virt/azure/driver.py:
```diff
@@ -0,0 +1,727 @@
+        Return the names of all the instances known to the virtualization
+        layer, as a list.
+        """
+        raise NotImplementedError()
```
Add a message while raising NotImplementedError. Same for everywhere

shahpratikr commented on nova/virt/azure/utils.py:
```diff
@@ -0,0 +1,335 @@
+
+
+def create_instance():
+    """Create GCE instance
```
its Azure

shahpratikr commented on requirements/requirements-aws.txt:
```diff
@@ -0,0 +1,2 @@
+boto>=2.32.1 # MIT
```
add boto3 as well.

shahpratikr committed:
Fixed service restart preferences in Omni devstack-plugin

Issue:
- When setting up devstack, Nova didn't find devstack service for
conductor. Hence set up was failing.

Fix:
- Removed unnecessary services from being restarted.

Change-Id: Ibdf16f50364ee53849b0a12726b9d9a17c9b0480
